src/common/types.ts

```diff
@@ -95,7 +96,7 @@ export interface IPlugin extends IRunnable {
   usableInDM?: boolean;
   usableInGuild?: boolean;
   validate(message: IMessage, args: string[]): boolean;
-  hasPermission(message: IMessage): true | string;
+  hasPermission(message: IMessage | discord.CommandInteraction): true | string;
```

Comment:
I thought you could put permissions in the configuration for slash commands? Then we don’t need to do our own validation ?

Reply:
Those validations are client-side validation checking the format. In our case we'd need to use lion's system for validation since it provides custom messages in chat for permissions errors.